test/e2e/far_e2e_test.go

```diff
@@ -71,61 +95,57 @@ var _ = Describe("FAR E2e", func() {
-				By("running fence_aws")
-			case configv1.BareMetalPlatformType:
-				fenceAgent = fenceAgentIPMI
-				nodeIdentifierPrefix = nodeIdentifierPrefixIPMI
-				By("running fence_ipmilan")
-			default:
-				Skip("FAR haven't been tested on this kind of cluster (non AWS or BareMetal)")
-			}
-
-			testShareParam, err := buildSharedParameters(clusterPlatform, fenceAgentAction)
-			if err != nil {
-				Fail("can't get shared information")
+			if secondRun {
+				nodeIndex++
 			}
-			testNodeParam, err := buildNodeParameters(clusterPlatform.Status.PlatformStatus.Type)
-			if err != nil {
-				Fail("can't get node information")
+			nodeName, errString = getNodeName(nodeIndex)
+			if errString != "" {
+				if nodeIndex <= 0 {
+					Fail(errString)
+				}
+				Skip(errString)
 			}
-			nodeName := v1alpha1.NodeName(testNodeName)
+			nodeNameParam := v1alpha1.NodeName(nodeName)
 			parameterName := v1alpha1.ParameterName(nodeIdentifierPrefix)
-			testNodeID = testNodeParam[parameterName][nodeName]
-			log.Info("Testing Node", "Node name", testNodeName, "Node ID", testNodeID)
+			testNodeID := testNodeParam[parameterName][nodeNameParam]
+			log.Info("Testing Node", "Node name", nodeName, "Node ID", testNodeID)
 
 			// save the node's boot time prior to the fence agent call
-			nodeBootTimeBefore, errBoot = e2eUtils.GetBootTime(clientSet, testNodeName, testNsName, log)
-			Expect(errBoot).ToNot(HaveOccurred(), "failed to get boot time of the node")
-
-			far = createFAR(testNodeName, fenceAgent, testShareParam, testNodeParam)
+			nodeBootTimeBefore, err = e2eUtils.GetBootTime(clientSet, nodeName, testNsName, log)
+			Expect(err).ToNot(HaveOccurred(), "failed to get boot time of the node")
+			far := createFAR(nodeName, fenceAgent, testShareParam, testNodeParam)
 			DeferCleanup(deleteFAR, far)
 		})
+		When("running FAR to reboot two nodes", func() {
+			It("should successfully remediate the first node", func() {
+				remediateNode(nodeName, succeesRebootMessage, nodeBootTimeBefore)
+				// next run create CR for the next worker node
+				secondRun = true
+			})
+			It("should successfully remediate the second node", func() {
+				remediateNode(nodeName, succeesRebootMessage, nodeBootTimeBefore)
 
-		When("running FAR to reboot node ", func() {
-			It("should execute the fence agent cli command", func() {
-				By("checking the CR has been created")
-				testFarCR := &v1alpha1.FenceAgentsRemediation{}
-				Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(far), testFarCR)).To(Succeed(), "failed to get FAR CR")
-
-				By("checking the command has been executed successfully")
-				checkFarLogs(succeesRebootMessage)
-
-				By("checking the node's boot time after running the FA")
-				wasNodeRebooted(testNodeName, nodeBootTimeBefore)
 			})
 		})
 	})
 })
 
+// getNodeName returns the node's name based on valid index, otherwise it returns an error
+func getNodeName(index int) (string, string) {
```

Comment:
Are you sure that nodes are returned in same order always, so that a different index always results in a different node?

Reply:
> otherwise it returns an error

I don't see an error, just 2 strings 🤷🏼‍♂️